Teuchos: Build warning in BigUInt

```
In file included from .../Trilinos/packages/teuchos/core/src/Teuchos_PrintDouble.cpp:43:
.../Trilinos/packages/teuchos/core/src/Teuchos_BigUInt.hpp:180:9: warning: variable 'xi' is used uninitialized whenever 'if' condition is false [-Wsometimes-uninitialized]
    if (i + ndigits < n) xi = x[i + ndigits] >> nbits;
        ^~~~~~~~~~~~~~~
.../Trilinos/packages/teuchos/core/src/Teuchos_BigUInt.hpp:254:10: note: in instantiation of member function 'Teuchos::BigUInt<34>::operator>>=' requested here
  factor >>= 1;
         ^
.../Trilinos/packages/teuchos/core/src/Teuchos_PrintDouble.cpp:193:27: note: in instantiation of function template specialization 'Teuchos::operator/<34>' requested here
      auto d_np1 = r_x_10 / s;
                          ^
.../Trilinos/packages/teuchos/core/src/Teuchos_BigUInt.hpp:182:12: note: uninitialized use occurs here
    x[i] = xi;
           ^~
.../Trilinos/packages/teuchos/core/src/Teuchos_BigUInt.hpp:180:5: note: remove the 'if' if its condition is always true
    if (i + ndigits < n) xi = x[i + ndigits] >> nbits;
    ^~~~~~~~~~~~~~~~~~~~~
.../Trilinos/packages/teuchos/core/src/Teuchos_BigUInt.hpp:179:21: note: initialize the variable 'xi' to silence this warning
    std::uint32_t xi;
                    ^
                     = 0
```

@trilinos/teuchos

--- a/packages/teuchos/core/src/Teuchos_BigUInt.hpp
+++ b/packages/teuchos/core/src/Teuchos_BigUInt.hpp
@@ -176,7 +176,7 @@ BigUInt<n>& BigUInt<n>::operator>>=(std::uint32_t b) {
   auto ndigits = b / 32;
   auto nbits = b - (ndigits * 32);
   for (int i = 0; i < n; ++i) {
-    std::uint32_t xi;
+    std::uint32_t xi = 0;
     if (i + ndigits < n) xi = x[i + ndigits] >> nbits;
     if (nbits && i + ndigits + 1 < n) xi |= x[i + ndigits + 1] << (32 - nbits);
     x[i] = xi;

